### PR TITLE
Add: store and restore the last selected workspace and space

### PR DESCRIPTION
**Pull request by Fraggle**
## Description

Keep the last workspace & space selected so it stop reverting to the first / global ones !

## Risk

None

## Deploy Plan

Deploy `front`

### CODE REVIEW

File: front/hooks/usePersistedNavigationSelection.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+export type NavigationSelectionType = {
+  lastWorkspaceId?: string;
+  lastSpaceId?: string;
+};
+
+export const NAVIGATION_SELECTION_COOKIE_NAME = "navigationSelection";
+
+export const usePersistedNavigationSelection = () => {
+  const [cookies, setCookie] = useCookies([NAVIGATION_SELECTION_COOKIE_NAME]);
```
Comment (flvndvd) [inline front/hooks/usePersistedNavigationSelection.ts:14]: Why not using local storage instead? IIRC, storing preferences in cookies isn't highly recommended.

My sense is that it it does not need to be secure and isn't needed by the SSR/API then local storage should be the go-to solution.

File: front/hooks/usePersistedNavigationSelection.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+  const setNavigationSelection = useCallback(
+    (selection: NavigationSelectionType) => {
+      const existingSelection = cookies.navigationSelection;
+
+      const newSelection = {
+        ...existingSelection,
+        ...selection,
+      } as NavigationSelectionType;
```
Comment (flvndvd) [inline front/hooks/usePersistedNavigationSelection.ts:23]:
```suggestion
      const newSelection: NavigationSelectionType = {
        ...existingSelection,
        ...selection,
      };
```

File: front/hooks/usePersistedNavigationSelection.ts (hunk 1 above)
Comment (flvndvd) [reply front/hooks/usePersistedNavigationSelection.ts:14]: Ok update: We actually use it on SSR, might be worth adding a comment here then to explain why cookies over local storage!

File: front/lib/persisted_navigation_selection.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+export const getPersistedNavigationSelection = (
+  cookies: NextApiRequestCookies
+): NavigationSelectionType => {
+  const selectionCookie = cookies[NAVIGATION_SELECTION_COOKIE_NAME];
+
+  if (!selectionCookie) {
+    return {};
+  }
+
+  try {
```
Comment (flvndvd) [inline front/lib/persisted_navigation_selection.ts:16]: We have `safeParseJSON`.

**Review by flvndvd — APPROVED** [review]

File: front/hooks/usePersistedNavigationSelection.ts (hunk 1 above)
Comment (Fraggle) [reply front/hooks/usePersistedNavigationSelection.ts:14]: You have a comment at the top explaining that it has a server-side counterpart :) (but okay no soucy)

**Comment by spolu** [comment]
Can we use `setUserMetadata` instead of a cookie :pray:

**Comment by Fraggle** [comment]
> Can we use setUserMetadata instead of a cookie 🙏

Yes, of course, I wasn't aware that we had it.

**Comment by spolu** [comment]
It is stateful so it's not exactly equivalent but it does serve the same purpose. Unless we have strong reasons to want a cookie, I'd be in favor of keeping our cookie footprint minimal.

**Comment by Fraggle** [comment]
I didn't want to create another model "just for that" so I went for cookie but my initial choice would have been stateful server side.

**Comment by spolu** [comment]
Then you already have a model. And even some legacy code: https://github.com/dust-tt/dust/pull/721 :)

**Pull request merged**